client/src/navigator/Drawer/SplitDrawerItemList.js

```diff
@@ -0,0 +1,159 @@
+import {
+  CommonActions,
+  DrawerActions,
+  useLinkBuilder,
+} from '@react-navigation/native'
+import { DrawerItem } from '@react-navigation/drawer'
+import React from 'react'
+import PropTypes from 'prop-types'
+
+/**
+ * A custom component that renders the two navigation lists in the drawer, with a
+ * child component in between. The purpose of this component is to create a navigation list of
+ * Contributor Courses followed by a navigation list of Learner Courses, with a Header and Container
+ * in between.
+ *
+ * Example: routes = ["abc", "123", "def"], firstRouteNames = ["abc"], secondRouteNames = ["123", "def"]
+ *
+ * Then the following will be returned:
+ * <DrawerItem route={"abc"}/>
+ * {middleChildComponent}
+ * <DrawerItem route={"123"}/>
+ * <DrawerItem route={"def"}/>
+ *
+ * @param state - React Navigation state, which contains the route data
+ * @param navigation - React Navigation, which contains navigation info
+ * @param descriptors - Contains styling information for the Drawer.Item
+ * @param firstRouteNames - contains the names of the routes that must appear in the first navigation list
+ * @param secondRouteNames - contains the names of the routes that must appear in the second navigation list
+ * @param middleChildComponent - the child component that should be rendered in between two navigation lists
+ *
+ * Original component: https://github.com/react-navigation/react-navigation/blob/main/packages/drawer/src/views/DrawerItemList.tsx
+ */
+const SplitDrawerItemList = ({
+  state,
+  navigation,
+  descriptors,
+  firstRouteNames,
+  secondRouteNames,
+  middleChildComponent,
+}) => {
+  const buildLink = useLinkBuilder()
+
+  const focusedRoute = state.routes[state.index]
+  const focusedDescriptor = descriptors[focusedRoute.key]
+  const focusedOptions = focusedDescriptor.options
+
+  console.log(focusedOptions)
+
+  /**
+   * Creates a DrawerItem given route data
+   * @param route = The route data
+   * @param i - The index of the route in the React Navigator
+   * @param zeroBasedShift - A shift to apply in the instance the route belongs to the second list
+   * @returns
+   */
+  const createCustomDrawerItem = (route, i, zeroBasedShift = 0) => {
```

Comment:
do u need to default zerobasedshift to 0 if 1 out of the 2 instances already has the prop passed in as 0